assets/js/truelayer-for-woocommerce.js

```diff
@@ -0,0 +1,51 @@
+/* global truelayerParams */
+jQuery(function ($) {
+    var truelayerForWooCommerce = {
+        Truelayer: null,
+
+        Payment: null,
+
+        init: function () {
+            window.addEventListener("hashchange", truelayerForWooCommerce.handleHashChange);
+            this.Truelayer = window.Truelayer;
+            this.Payment = this.Truelayer?.Payment;
+        },
+
+        handleHashChange: function() {
+            var currentHash = location.hash;
+            var splittedHash = currentHash.split(":");
+            if( splittedHash[0] === "#truelayer" ){
+                var url = atob( splittedHash[1] );
+                var payment_id = atob( splittedHash[2] );
+                var payment_token = atob( splittedHash[3] );
+                truelayerForWooCommerce.initializeEpp(url, payment_id, payment_token);
+            }
+        },
+
+        initializeEpp: function( url, payment_id, payment_token ) {
+            const payment = this.Payment({
```

Comment:
Maybe add a check for if `this.Payment` is null. Since there is a risk that it might be incase the Truelayer SDK did not load in properly. If we don't do that there might be a case where we cause a error that is not handled.